pkg/kds/server/server_test.go

```diff
@@ -0,0 +1,628 @@
+	fi1 = mesh_proto.FaultInjection{
+		Sources: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Destinations: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Conf: &mesh_proto.FaultInjection_Conf{
+			Abort: &mesh_proto.FaultInjection_Conf_Abort{
+				Percentage: &wrappers.DoubleValue{Value: 90},
+				HttpStatus: &wrappers.UInt32Value{Value: 404},
+			},
+		},
+	}
+	ingress = mesh_proto.Dataplane{
+		Networking: &mesh_proto.Dataplane_Networking{
+			Ingress: &mesh_proto.Dataplane_Networking_Ingress{
+				AvailableServices: []*mesh_proto.Dataplane_Networking_Ingress_AvailableService{{
+					Tags: map[string]string{
+						"service": "backend",
+					}},
+				},
+			},
+			Address: "192.168.0.1",
+		},
+	}
+	cb1 = mesh_proto.CircuitBreaker{
+		Sources: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Destinations: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Conf: &mesh_proto.CircuitBreaker_Conf{
+			Detectors: &mesh_proto.CircuitBreaker_Conf_Detectors{},
+		},
+	}
+	hc1 = mesh_proto.HealthCheck{
+		Sources: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Destinations: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Conf: &mesh_proto.HealthCheck_Conf{
+			ActiveChecks: &mesh_proto.HealthCheck_Conf_Active{
+				Interval: &duration.Duration{Seconds: 5},
+				Timeout:  &duration.Duration{Seconds: 7},
+			},
+		},
+	}
+	tl1 = mesh_proto.TrafficLog{
+		Sources: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Destinations: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Conf: &mesh_proto.TrafficLog_Conf{
+			Backend: "logging-backend",
+		},
+	}
+	tp1 = mesh_proto.TrafficPermission{
+		Sources: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Destinations: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+	}
+	tr1 = mesh_proto.TrafficRoute{
+		Sources: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Destinations: []*mesh_proto.Selector{{
+			Match: map[string]string{
+				"service": "*",
+			},
+		}},
+		Conf: []*mesh_proto.TrafficRoute_WeightedDestination{{
+			Weight: 10,
+			Destination: map[string]string{
+				"version": "v2",
+			},
+		}},
+	}
+	tt1 = mesh_proto.TrafficTrace{
+		Selectors: []*mesh_proto.Selector{{
+			Match: map[string]string{"serivce": "*"},
+		}},
+		Conf: &mesh_proto.TrafficTrace_Conf{
+			Backend: "tracing-backend",
+		},
+	}
+	pt1 = mesh_proto.ProxyTemplate{
+		Selectors: []*mesh_proto.Selector{{
+			Match: map[string]string{"serivce": "*"},
+		}},
+		Conf: &mesh_proto.ProxyTemplate_Conf{
+			Imports: []string{"default-kuma-profile"},
+		},
+	}
```

Comment:
The painful bit here is that we don't automagically get this tested once we add a new type. All the tests will just pas without even noticing the new resource type. I guess that's the best we can do today though.